src/element.rs

```diff
@@ -37,5 +37,6 @@ pub trait ElementDatabase
 pub trait ElementFilesystem
 {
 	fn insert_into(&self, fs: &Filesystem) -> Result<(), Error>;
+	fn create(&mut self, fs: &Filesystem) -> Result<(), Error>;
```

Comment:
I would suggest if something doesn't apply to every element, we don't put it into the Element trait. Or we create a new trait that apply only to containers (Collections, albums)